docs-dev/project/template/implementation-plan.md

```diff
@@ -0,0 +1,51 @@
+# Implementation Plan Template
```

Comment:
we should always have high level plan and ui ( dsl in contenxt of ruby lib ) and maybe responsibility and what libs should we use and what for 

why do we have two templates - in guides and in project - we should merge it